mu/integration_tests/test_babydriver.py

```diff
@@ -0,0 +1,88 @@
+import time
+import unittest
+import sys
+sys.path.append('projects/baby_driver/scripts')
+# pylint: disable=wrong-import-position
+import repl_setup
+import gpio_set
+import i2c_write
+import spi_exchange
+import gpio_interrupts
+import gpio_get
+import can_util
+import adc_read
+from mu.integration_tests import int_test
+
+
+
+class TestBabyDriver(int_test.IntTest):
+    def setUp(self):
+        super().setUp()
+        self.babydriver = self.manager.start('baby_driver')
+        can_util.default_channel = "vcan0"
+        repl_setup.initialize_modules()
+
+    def test_adc_read(self):
+        time.sleep(1)
+        adc_read.adc_read(port='A', pin=6, raw=False)
+
+    def test_gpio_get(self):
+        time.sleep(1)
+        self.babydriver.set_gpio(port='A', pin=5, state=False)
+        time.sleep(1)
+        assert gpio_get.gpio_get('A', 5) is False
+
+    def test_gpio_set(self):
+        time.sleep(1)
+        self.babydriver.set_gpio(port='A', pin=5, state=False)
+        time.sleep(1)
+        gpio_set.gpio_set('A', 5, True)
+        time.sleep(1)
+        assert self.babydriver.get_gpio(port='A', pin=5) is True
+
+    def test_gpio_interrupts(self): #IN PROGRESS
+        # Defining a Callback function
+        # pylint: disable=attribute-defined-outside-init
+        isCalled = False
+        def interrupt_callback(info):
+            self.isCalled = True
+        # Registring the interrupt
+        time.sleep(1)
+        gpio_interrupts.register_gpio_interrupt(port='A',
+                                                pin=5,
+                                                callback =interrupt_callback(info="info"))
+        # Simulating
+        time.sleep(1)
+        self.babydriver.set_gpio(port='A', pin=5, state=True)
+        time.sleep(1)
+        self.babydriver.set_gpio(port='A', pin=5, state=False)
+        # Asserting
+        time.sleep(1)
+        assert isCalled is True
+        # Unregistring the interrupt
+        time.sleep(1)
```

Comment:
Just to speed it up a notch, I don't think you need this particular `time.sleep` call since the assert shouldn't have done anything that needs to be waited for, so you can probably remove it. (But if things start failing, feel free to put it back.)